pkg/reconciler/route/resources/ingress.go

```diff
@@ -253,8 +269,12 @@ func makeBaseIngressPath(ns string, targets traffic.RevisionTargets) *netv1alpha
 		})
 	}
 
+	var timeout *metav1.Duration
+	if sawDuration {
+		timeout = &metav1.Duration{Duration: duration}
+	}
```

Comment:
btw, rather than having two variables you can just have `timeout` above, initialized to `nil` and then below where you set `sawDuration` initialize/update it?